File: leetcode_75/HashMap_Set/cpp/1207.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
#include <unordered_set>

using namespace std;

class Solution
{
public:
    bool uniqueOccurrences(vector<int> &arr)
    {
        unordered_map<int, int> m;
        for (auto num : arr)
        {
            if (m.count(num) == 0)
            {
                m[num] = 0;
            }
            m[num] += 1;
        }

        std::vector<int> values;
        for (auto const &[key, value] : m)
        {
            values.push_back(value);
        }

        unordered_set<int> d(values.begin(), values.end());
        if (values.size() == d.size())
        {
            return true;
        }
        return false;
    }
};

File: leetcode_75/HashMap_Set/python/1207.py
from typing import List


class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        size = len(arr)
        d = dict()
        for i in range(size):
            if arr[i] not in d:
                d[arr[i]] = 0
            d[arr[i]] += 1

        values = d.values()
        if len(set(values)) != len(values):
            return False
        return True


solution = Solution()
print(solution.uniqueOccurrences(arr=[-3, 0, 1, -3, 1, 1, 1, -3, 10, 0]))
